File: two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_table = {}
        for index, val in enumerate(nums):
            if not val in hash_table.keys():
                hash_table[val] = [index]
            else:
                hash_table[val].append(index)
        for val in nums:
            compliment = target-val
            if compliment == val and len(hash_table[val])>=2:
                return hash_table[val][:2]
            if compliment in hash_table.keys() and compliment != val:
                return [hash_table[val][0], hash_table[compliment][0]]
            

File: easy/two-sum.cpp

#include<bits/stdc++.h>;

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,vector <int>> m;
        vector<int> results;
        
        for (int i = 0; i < nums.size(); i++){
            
            if (m.find(nums[i]) == m.end()) {
                vector <int> arr;
                arr.push_back(i);
                m.insert(make_pair(nums[i], arr));
            }
            
            else{
                m[nums[i]].push_back(i);
            }
        }
        
        for(int i = 0; i< nums.size(); i++ ){
            
            int to_check = target - nums[i];
            
            if (m.find(to_check) != m.end() && ((to_check != nums[i]) || (to_check == nums[i] && m[to_check].size() > 1))) {        
                    results.push_back(i);
                    results.push_back(m[to_check].back());
                    break;
            }
        }
        return results;
    }
};